cli/monkey-cam.cpp

```diff
@@ -97,10 +97,15 @@ std::unique_ptr<BoardShape> loadBoard(boost::property_tree::ptree& config) {
     tailPack = loadInserts(tpc.get());
   }
 
+  auto noseEdgeExt = config.get_optional<double>("board.nose edge extension")
+    .get_value_or(999.0);
```

Comment:
Just do `auto noseEdgeExt = config.get_optional<double>("blah");`. That will make `noseEdgeExt` of type `boost::optional<double>`. Pass that along to the `BoardShape` constructor, and make the new members of type `boost::optional<double>`. Then later, instead of checking against 999.0, I think you can do `if (m_foo)`, and use `*m_foo` when you know it has a value. I've never really played with Boost.Optional, but it should give us what we want here without using magic numbers. See http://www.boost.org/doc/libs/1_47_0/libs/optional/doc/html/index.html for more details. Give it a shot and let's see how it works out.